### PR TITLE
[LP#2039667] Provide charm config parameters on the default storage class

### PR DESCRIPTION
**Pull request by addyess**
[LP#2039667](https://bugs.launchpad.net/charm-vsphere-cloud-provider/+bug/2039667)

Allows the user to configure arbitrary parameters on the default storage class installed

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,7 +11,6 @@ show_missing = true
 [tool.pytest.ini_options]
 minversion = "6.0"
 log_cli_level = "INFO"
-asyncio_mode = "auto"
```
Comment (addyess) [inline pyproject.toml:14]: removes a testing warning

File: src/charm.py [hunk 2]
```diff
@@ -217,7 +217,7 @@ def _install_or_upgrade(self, event, config_hash=None):
                 controller.apply_manifests()
             except ManifestClientError as e:
                 self.unit.status = WaitingStatus("Waiting for kube-apiserver")
-                log.warn(f"Encountered retryable installation error: {e}")
+                log.warning(f"Encountered retryable installation error: {e}")
```
Comment (addyess) [inline src/charm.py:220]: removes a deprecated method call

File: upstream/update.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def __hash__(self) -> int:
         return hash(self.name)
 
     def __eq__(self, other) -> bool:
-        """Comparible based on its name."""
+        """Comparable based on its name."""
```
Comment (addyess) [inline upstream/update.py:91]: fixes misspelling

**Review by mateoflorido — APPROVED** [review]
LGTM ⭐

**Pull request merged**